File: src/main/tests/input/class.cpp
#include <iostream>
#include <string>


class Person {
private:
    std::string name;
    int age;

public:

    Person(const std::string& name, int age) {
        this->name = name;
        this->age = age;
    }

    std::string getName() const {
        return this->name;
    }

    void setName(const std::string& name) {
        this->name = name;
    }

    int getAge() const {
        return this->age;
    }

    void setAge(int newAge) {
        if (newAge >= 0) {
            this->age = newAge;
        }
    }
    void display(){
        std::cout << "Name: " << this->name << ", age: " << this->age <<std::endl;
    }

};

int main() {
    Person *p1= new Person("Neko",25);
    p1->display();

    Person p2("Alice", 30);
    p2.display();

    p2.setName("Bob");
    p2.setAge(35);
    p2.display();

    return 0;
}

File: src/main/tests/pythonGenerated/class.py
import typing

class Person:
    def __init__(self, name, age):
        self.name = name
        self.age = age
    
    def getName(self) -> str:
        return self.name
    
    def setName(self, name):
        self.name = name
    
    def getAge(self) -> int:
        return self.age
    
    def setAge(self, newAge):
        if newAge >= 0:
            self.age = newAge
    
    def display(self):
        print("Name: ", self.name, ", age: ", self.age, sep="")
        print()

def main() -> int:
    p1 = Person("Neko", 25)
    p1.display()
    p2 = Person("Alice", 30)
    p2.display()
    p2.setName("Bob")
    p2.setAge(35)
    p2.display()
    return 0

if __name__ == "__main__":
    main()
